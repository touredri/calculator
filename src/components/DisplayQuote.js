import { useState, useEffect } from 'react';

const DisplayQuote = () => {
  const [quote, setQuote] = useState([]);
  const [isloading, setLoading] = useState(false);
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    setLoading(true);
    const fetchData = async () => {
      setLoading(true);
      try {
        const res = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=love',
          {
            headers: {
              'X-Api-Key': 'nA8aOYlbqZZxy+CR4vGCaQ==nFOrV3FMpzOrHcnO',
            },
          },
        );
        const data = await res.json();
        setQuote([data[0].quote, data[0].author]);
        setLoading(false);
      } catch (error) {
        setHasError(true);
      }
      setLoading(false);
    };
    fetchData();
  }, []);
  if (isloading) return <div> Wait a moment....</div>;
  if (hasError) return <div>Oups something wrong!!</div>;

  return (
    <div>
      <div>Quote of the day</div>
      <p>{quote[0]}</p>
      <p>
        -
        {quote[1]}
      </p>
    </div>
  );
};

export default DisplayQuote;
