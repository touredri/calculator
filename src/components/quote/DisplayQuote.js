import { useState, useEffect } from 'react';
import '../calculator.css';

const DisplayQuote = () => {
  const [quote, setQuote] = useState([]);
  const [isloading, setLoading] = useState(false);
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    setLoading(true);
    const fetchData = async () => {
      setLoading(true);
      try {
        const res = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=love',
          {
            headers: {
              'X-Api-Key': 'nA8aOYlbqZZxy+CR4vGCaQ==nFOrV3FMpzOrHcnO',
            },
          },
        );
        const data = await res.json();
        setQuote([data[0].quote, data[0].author]);
        setLoading(false);
      } catch (error) {
        setHasError(true);
      }
      setLoading(false);
    };
    fetchData();
  }, []);
  if (isloading) return <div className="wait"> Wait a moment....</div>;
  if (hasError) return <div>Oups something wrong!!</div>;

  return (
    <div
      className="quote"
      style={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'column',
        width: '100%',
      }}
    >
      <div style={{ textAlign: 'center', color: 'black' }}>Quote of the day</div>
      <p style={{ textAlign: 'center' }}>
        &quot;
        {quote[0]}
        &quot;
      </p>
      <p>
        -
        {quote[1]}
      </p>
    </div>
  );
};

export default DisplayQuote;
