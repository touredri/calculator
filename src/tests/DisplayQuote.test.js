import { render } from '@testing-library/react';
import DisplayQuote from '../components/quote/DisplayQuote';

describe('DisplayQuote', () => {
  test('should render correctly', () => {
    const { asFragment } = render(<DisplayQuote />);
    expect(asFragment()).toMatchSnapshot();
  });
});
