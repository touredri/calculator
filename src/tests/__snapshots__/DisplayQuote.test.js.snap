// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayQuote should render correctly 1`] = `
<DocumentFragment>
  <div
    class="wait"
    style=""
  >
     Wait a moment....
  </div>
</DocumentFragment>
`;
